modules/imgcodecs/src/grfmt_gif.cpp

```diff
@@ -666,7 +682,7 @@ bool GifEncoder::writeFrame(const Mat &img) {
     strm.putByte(0xF9); // graphic control label
     strm.putByte(0x04); // block size, fixed number
     // flag is a packed field, and the first 3 bits are reserved
-    uchar flag = opMode << 2;
+    uchar flag = GifDisposeMethod::GIF_DISPOSE_NONE << 2;
```

Comment:
`uchar flag = GifDisposeMethod::GIF_DISPOSE_RESTORE_PREVIOUS << 2;`

Reply:
I'm sorry I changed it to use new enum, but I missed to use restoreToPrevious method. I fixed it.

Reply:
M.b. stupid question: why do you use flag values 0,1,2, and then shift left / shift right them during IO. The flag values may be shifted here once. Also please replace & 0x3 with some meaningful constant name like `GIF_DISPOSE_FLAG_MASK`.

Reply:
I fixed around the current Disposal Method, thank you for your point !